stock_available_product_expiry/reports/stock_scrap_expired.xml

```diff
@@ -0,0 +1,101 @@
+                                    <td t-if="o.origin">
+                                        <span t-field="o.origin"/>
+                                    </td>
+                                    <td>
+                                        <span t-field="o.state"/>
+                                    </td>
+                                    <td>
+                                        <span t-field="o.removal_date"/>
+                                    </td>
+                                </tr>
+                            </tbody>
+                        </table>
+                        <br/>
+                        <br/>
+                        <table class="table table-condensed" t-if="o.stock_scrap_expired_line_ids">
+                            <t t-set="lines" t-value="o.stock_scrap_expired_line_ids"/>
+                            <t t-set="has_barcode" t-value="any([move.product_id and move.product_id.sudo().barcode for move in lines.mapped('move_id')])"/>
+                            <t t-set="has_serial_number" t-value="lines.filtered('lot_id')" groups="stock.group_production_lot"/>
+                            <thead>
+                                <tr>
+                                    <th><strong>Product</strong></th>
+                                    <th class="text-right"><strong>Quantity</strong></th>
+                                    <th><strong>Source</strong></th>
+                                    <th t-if="has_barcode" class="text-center">
+                                        <strong>Barcode</strong>
+                                    </th>
+                                    <th><strong>Lot/Serial Number</strong></th>
+                                    <th><strong>Expiry date</strong></th>
+                                </tr>
+                            </thead>
+                            <tbody>
+                                <tr t-foreach="lines" t-as="line">
+                                    <td><span t-field="line.product_id"/><br/>
+                                        <span t-field="line.product_id.description_picking"/>
+                                        <t t-if="not line.product_id and line.package_id"><span  t-field="line.package_id"/></t></td>
+                                    <td class="text-right">
+                                        <span t-field="line.scrap_qty"/>
+                                    </td>
+                                    <td>
+                                        <span t-field="line.location_id"/>
+                                    </td>
+                                    <td t-if="has_barcode">
+                                        <span t-if="line.product_id.barcode">
+                                            <img t-att-src="'/report/barcode/?type=%s&amp;value=%s&amp;width=%s&amp;height=%s' % ('Code128', line.product_id.barcode, 600, 100)" style="width:300px;height:50px"/>
```

Comment:
@acsonefho Should be EAN13. See https://github.com/odoo/odoo/blob/10.0/addons/product/report/product_product_templates.xml#L20